catalog/core/catalog-core-actions/src/main/java/org/codice/ddf/catalog/actions/AbstractMetacardActionProvider.java

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.actions;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.codice.ddf.configuration.SystemInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.action.Action;
+import ddf.action.ActionProvider;
+import ddf.catalog.data.Metacard;
+
+/**
+ * Abstract base class for {@link ActionProvider}s that can handle {@link Metacard} source objects.
+ */
+public abstract class AbstractMetacardActionProvider implements ActionProvider {
+
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(AbstractMetacardActionProvider.class);
+
+    private static final String UNKNOWN_TARGET = "0.0.0.0";
+
+    private final String actionProviderId;
+
+    private final String title;
+
+    private final String description;
+
+    /**
+     * Constructor that accepts the values to be used when a new {@link Action} is created by this
+     * {@link ActionProvider}.
+     *
+     * @param actionProviderId ID that will be assigned to the {@link Action} that will be
+     *                         created. Cannot be empty or blank.
+     * @param title            title that will be used when this {@link ActionProvider} creates a
+     *                         new {@link Action}
+     * @param description      description that will be used when this {@link ActionProvider}
+     *                         creates a new {@link Action}
+     */
+    protected AbstractMetacardActionProvider(String actionProviderId, String title,
+            String description) {
+        Validate.notNull(actionProviderId, "Action provider ID cannot be null");
+        Validate.notEmpty(actionProviderId.trim(), "Action provider ID cannot be empty or blank");
```

Comment:
Validate has a notBlank() method that will check if the string is null, zero-length, or only whitespaces - that way it will show your error message if actionProviderId is null instead of just throwing a nullPointerException on trim().